core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -111,12 +113,19 @@ private[spark] class BlockManager(
 
   // Client to read other executors' shuffle files. This is either an external service, or just the
   // standard BlockTransferService to directly connect to other Executors.
-  private[spark] val shuffleClient = if (externalShuffleServiceEnabled) {
-    val transConf = SparkTransportConf.fromSparkConf(conf, numUsableCores)
-    new ExternalShuffleClient(transConf, securityManager, securityManager.isAuthenticationEnabled(),
-      securityManager.isSaslEncryptionEnabled())
-  } else {
-    blockTransferService
+  private[spark] val shuffleClient = mockShuffleClient.getOrElse(createShuffleClient)
+
+  private def createShuffleClient: ShuffleClient = {
```

Comment:
I think it's more readable this way, so unless you feel strongly about it I'd leave it as such.